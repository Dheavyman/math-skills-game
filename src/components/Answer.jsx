import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  selectedNumbers: PropTypes.arrayOf(PropTypes.number).isRequired,
  handleUnselectNumber: PropTypes.func.isRequired
};

/**
 * Answer component
 *
 * @param {object} props - Properties passed to component
 *
 * @returns {object} React element
 */
const Answer = (props) => {
  const { selectedNumbers, handleUnselectNumber } = props;
  return (
    <div className="col-5">
      <div className="number">
        {selectedNumbers.map((number, index) => (
          <span
            role="button"
            tabIndex="0"
            key={index.toString()}
            onClick={() => handleUnselectNumber(number)}
            onKeyUp={() => {}}
          >
            {number}
          </span>
        ))}
      </div>
    </div>
  );
};

Answer.propTypes = propTypes;

export default Answer;
