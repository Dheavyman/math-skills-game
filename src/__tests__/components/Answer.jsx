import React from 'react';
import TestRenderer from 'react-test-renderer';

import Answer from '../../components/Answer';

describe('Answer component', () => {
  const props = {
    selectedNumbers: [2, 7, 9],
    handleUnselectNumber: jest.fn()
  };

  it('should render component successfully', () => {
    const renderer = TestRenderer.create(<Answer {...props} />);

    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it(`should call handle unselect number when a selected number is
    clicked`, () => {
    const renderer = TestRenderer.create(<Answer {...props} />);
    renderer.root.findAllByType('span')[0].props.onClick();
    renderer.root.findAllByType('span')[2].props.onKeyUp({ key: 'Enter' });
    expect(props.handleUnselectNumber).toHaveBeenCalledWith(2);
    expect(props.handleUnselectNumber).toHaveBeenLastCalledWith(9);
  });
});
