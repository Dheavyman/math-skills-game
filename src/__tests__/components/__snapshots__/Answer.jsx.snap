// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Answer component should render component successfully 1`] = `
<div
  className="col-5"
>
  <div
    className="number"
  >
    <span
      className="answer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      2
    </span>
    <span
      className="answer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      7
    </span>
    <span
      className="answer"
      onClick={[Function]}
      onKeyUp={[Function]}
      role="button"
      tabIndex="0"
    >
      9
    </span>
  </div>
</div>
`;
